docs/protocol-basics.rst

```diff
@@ -138,8 +136,60 @@ block public key.
 
 .. _status:
 
-Status
-------
+Status (protocol 2.0 and later)
+-------------------------------
+
+To calculate the `status` of a :ref:`script hash <script hashes>` (or
+address):
+
+1. Consider all transactions touching the script hash (both those spending
+from it, and those funding it), both confirmed and unconfirmed (in mempool).
+
+2. Order confirmed transactions by increasing height (and position in the
+block if there are more than one in a block).
```

Comment:
The script hash status is dependent on the transaction ordering.
If I order transactions incorrectly, I get the wrong script status hash and I'll do some extra requests which increase the server load.
This is more likely to happen if I have multiple transactions involving a script in the same block.

How the client should know the (relative) position of a transaction in the block?
Should the client infer it from the order in the returned list by previous calls to `get_history`?

Does it make sense to use the same ordering that is used for mempool transactions for confirmed transactions too?

AFAICT it would be simpler for client libraries to implement correctly, and indirectly it should help the server too.